src/server/tran_server.hpp

```diff
@@ -72,21 +72,62 @@ class tran_server
 
     int boot (const char *db_name);
 
-    void disconnect_page_server ();
-    bool is_page_server_connected () const;
-    bool is_page_server_connected (const size_t idx) const;
-    void push_request_to (size_t idx, tran_to_page_request reqid, std::string &&payload);
+    /* send request to a specific connection */
+    void push_request (size_t idx, tran_to_page_request reqid, std::string &&payload);
+    int send_receive (size_t idx, tran_to_page_request reqid, std::string &&payload_in, std::string &payload_out) const;
```

Comment:
These two functions not used publicly. Consider making them private. To be exposed publicly if/when needed.

Reply:
Good. Done.